### DISCUSSION TITLE
Prisma TypeError: Cannot read properties of undefined (reading 'findMany')

### DISCUSSION THREAD

**Discussion by firaki12345-cmd** [prisma/prisma#22648] [Q&A] [answered]
### Question

so i already had a postgres database with a table and some data in it

i followed the prisma instruction on how to add an existing database with prisma and configured it succesfully, but then when i want to get all the data, i get this error in the terminal.

**My Prisma Version Details**

```
prisma                  : 5.8.0
@prisma/client          : 5.8.0
Computed binaryTarget   : debian-openssl-3.0.x
Operating System        : linux
Architecture            : x64
Node.js                 : v20.5.1
```

**my prisma scehma below:**

```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model retailer {
  id          Int      @id @default(autoincrement())
  links       String   @db.VarChar(9999)
  shop        String   @db.VarChar(100)
  country     String   @db.VarChar(300)
  category    String   @db.VarChar(300)
  title       String   @db.VarChar(300)
  price       Decimal  @db.Decimal(10, 2)
  image       String   @db.VarChar(100)
  color       String   @db.VarChar(300)
  slug        String   @db.VarChar(50)
  date_posted DateTime @db.Timestamptz(6)
  no_of_likes Int
}
```

**my client.jsx file**

```
import { PrismaClient } from "@prisma/client"

const prismaClientSingleton = () => { 
return new PrismaClient() 
}

const prisma = globalThis.prisma ?? prismaClientSingleton()

export default prisma

if (process.env.NODE_ENV !== "production") globalThis.prisma = prisma
```

**my route.jsx file**

```
import { NextRequest, NextResponse } from "next/server";
import prisma from '@prisma/client'

export async function GET(NextRequest){
  console.log(prisma)
  const data = await prisma.retailer.findMany()
  console.log(data)
  return NextResponse.json("hello")
}
```

**Error Below**

```
⨯ TypeError: Cannot read properties of undefined (reading 'findMany')
    at GET (webpack-internal:///(rsc)/./app/api/products/route.jsx:20:86)
```

**Please help**

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment, marked as answer]
Hello @firaki12345-cmd 👋 

It looks like you are importing prisma incorrectly. You have instantiated the PrismaClient in `client.jsx` file but you are not using it in the `route.tsx` file.

Can you try changing the import in `route.jsx` from
```ts
import prisma from '@prisma/client'
```
to
```ts
import prisma from '../../path/to/your/client'
```
Please make sure to replace ../../path/to/your/client with the actual relative path to your `client.jsx` file.

**Reply by firaki12345-cmd** [reply]
thanks you so much. i actully started from scratch and installed prisma and model and everything, and your solution worked.. i was so stupid

`import prisma from '../../../prisma/client.jsx'`

**GOD BLESS YOU**

**Reply by nurul3101** [reply]
I am glad to hear that the solution worked! 🙏
